src/api/posts/src/storage.js

```diff
@@ -22,6 +22,26 @@ const createInvalidFeedKey = (id) => createFeedKey(id).concat(invalidSuffix);
 // "NirlSYranl" to "t:feed:NirlSYranl:delayed"
 const createDelayedFeedKey = (id) => createFeedKey(id).concat(delayedSuffix);
 
+const getInvalidKeys = async () => {
+  let keysArray = [];
+  const stream = redis.scanStream({
+    match: 't:feed:*:invalid',
+  });
+  return new Promise((res, rej) => {
+    stream.on('data', async (keys) => {
+      keysArray = keysArray.concat(keys);
+    });
+    stream.on('end', () => {
+      //remove duplicate keys
```

Comment:
There won't be any, since keys are unique.  You can skip this bit and just `resolve` the array.

Reply:
Hi prof, about this , in the ioredis documentation , they mention that keys might contain duplicate values due to Scan's implementation in redis.
example: 
```
const redis = new Redis();
// Create a readable stream (object mode)
const stream = redis.scanStream();
stream.on("data", (resultKeys) => {
  // `resultKeys` is an array of strings representing key names.
  // Note that resultKeys may contain 0 keys, and that it will sometimes
  // contain duplicates due to SCAN's implementation in Redis.
  for (let i = 0; i < resultKeys.length; i++) {
    console.log(resultKeys[i]);
  }
});
stream.on("end", () => {
  console.log("all keys have been visited");
});
``` 
I am new to Redis and ioredis so  I am not sure about this issue ?

Reply:
OK, if that's the case, I would use a `Set()` vs. an `Array` to store these, and then you can [`.add()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add) as many times as you want and we won't get duplicates.  When it's done (i.e., in `"end"`), you can give back `[...set]` or `Array.from(set)` or something to convert it to an Array.

Reply:
Thank you, I have made a change according to your suggestion.